cli/mender-artifact/main.go

```diff
@@ -448,9 +452,58 @@ func getCliContext() *cli.App {
 	}
 	app.Flags = append([]cli.Flag{}, globalFlags...)
 
+	cli.HelpPrinter = upgradeHelpPrinter(cli.HelpPrinter)
+
 	return app
 }
 
 func run() error {
 	return getCliContext().Run(os.Args)
 }
+
+func upgradeHelpPrinter(defaultPrinter func(w io.Writer, templ string, data interface{})) func(
+	w io.Writer, templ string, data interface{}) {
+	// Applies the ordinary help printer with column post processing
+	return func(stdout io.Writer, templ string, data interface{}) {
+		var buf bytes.Buffer
+		defaultPrinter(&buf, templ, data)
+		const alpha = "abcdefghijklmnopqrstuvxyz"
+		terminalWidth, _, err := terminal.GetSize(int(os.Stdin.Fd()))
+		if err != nil {
+			// Just write help as is.
+			stdout.Write(buf.Bytes())
+			return
+		}
+		for line, err := buf.ReadString('\n'); err == nil; line, err = buf.ReadString('\n') {
+			if len(line) <= terminalWidth+1 {
+				stdout.Write([]byte(line))
+				continue
+			}
+			newLine := line
+			indent := strings.LastIndex(
+				line[:terminalWidth], "  ")
+			indent += strings.IndexAny(strings.ToLower(line[indent:]),
+				alpha) - 1
+			// find indentation of last column
+			if indent >= terminalWidth-1 ||
+				indent == -1 {
+				indent = 0
+			}
```

Comment:
I think the `line[indent:]` will panic if the first expression results in `-1`. This can happen for very narrow terminals. So that needs to be checked immediately after the first expression.

Also I think the second expression will not work correctly if the first characters are not alphanumeric. I don't fully understand why it is needed at all, but would it be better to `TrimLeft` with spaces as the set?

Reply:
This was just intended for a special case for the mender help menu.
The addition to the indent aligns text that are prefixed with non-alphanumeric characters such as enumerations, e.g.
```
Lorem ipsum etc.
     -- this is a list entry and should continue at
        this indentation level, not
     here.
```

Reply:
Ah, ok, smart!

My other point is still valid though.

Reply:
oops.. I only made the changes for the client PR, and forgot to update this one. I did some minor changes to improve readability as well.